src/plugins/discord/address.js

```diff
@@ -0,0 +1,56 @@
+// @flow
+
+import type {NodeAddressT, EdgeAddressT} from "../../core/graph";
+import {NodeAddress, EdgeAddress} from "../../core/graph";
+import {type Snowflake} from "./models";
+import {
+  userNodeType,
+  messageNodeType,
+  sentMessageEdgeType,
+} from "./declaration";
+
+/**
+ * These user IDs are considered globally unique by Discord.
+ * We can reference a user even across multiple guilds using this address.
+ * See https://discordapp.com/api/users/{userId}
+ *
+ * Importantly we don't include any membership details in the address.
+ * This way the graph topology stays identical when users leave / rejoin.
+ */
+export function userAddress(userId: Snowflake): NodeAddressT {
+  return NodeAddress.append(userNodeType.prefix, userId);
+}
+
+/**
+ * The URLs Discord creates to permalink these are:
+ * https://discordapp.com/channels/{guildId}/{channelId}/{messageId}
+ * Meanwhile the API retrieves these using:
+ * https://discordapp.com/api/channels/{channelId}/messages/{messageId}
+ *
+ * In other words the API finds just {channelId} sufficiently namespaced for unique messages.
+ * They're being conservative for the permalinks by namespacing as {guildId}/{channelId}.
+ * As they've clearly not felt the need to change the API, we'll use just {channelId} too.
```

Comment:
One consideration would be SC instances that have multiple Discord servers in their configuration and want to set different weights for each one. For that to happen, we might need to include `guildId` in the Address structure.

Reply:
Multiple servers can theoretically be supported even when we don't include the `guildId` here. This is because the `channelId` Snowflake is considered sufficiently unique to be in a global namespace.

But I think your point is, we can't express this in a single address prefix to include all messages in a particular server. So to do something like 2x on all messages in a particular server, we'd need to set this 2x on a more specific prefix, which is for each of it's channels.

A fair point. Though I think we shouldn't change our addresses to suit the weighting-by-prefix system. There's a couple of issues with the current weight configuration, so we'll likely want to have a more flexible weight configuration approach.

For example, if we want to set different weights in GitHub between open / closed / merged PRs (#1682) we'd be forced to include this (open/closed/merged) mutable piece of state into the node address to do this with prefix weights... which isn't the purpose of addresses, they're unique IDs.

Because we're also not supporting multiple servers yet, I think it's safe to punt on this :]